### DISCUSSION TITLE
Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.

### DISCUSSION THREAD

**Discussion by clairelovesgravy** [openai/whisper#1218] [Q&A] [answered]
It seems I have installed whisperai successfully, but when I run it to transform audio, it doesn't work. I have searched the question and then tried 
reinstall, restart, but it keeps on giving the following error:
claire@claires-MacBook-Pro audio % whisper demo.m4a
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/__init__.py:63: UserWarning: /Users/claire/.cache/whisper/small.pt exists, but the SHA256 checksum does not match; re-downloading the file
  warnings.warn(
100%|██████████████████████████████████████▉| 460M/461M [02:40<00:00, 3.00MiB/s]
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/bin/whisper", line 8, in <module>
    sys.exit(cli())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/transcribe.py", line 433, in cli
    model = load_model(model_name, device=device, download_root=model_dir)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/__init__.py", line 131, in load_model
    checkpoint_file = _download(_MODELS[name], download_root, in_memory)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/__init__.py", line 85, in _download
    raise RuntimeError(
RuntimeError: Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.

Is there an enthusiastic person who can help me fix the problem? salite!

**Comment by jongwook** [comment, marked as answer]
Could you run this:

```
ls -al /Users/claire/.cache/whisper/small.pt
```

and paste the result?

It seems possible that due to a network issue the file is not getting fully downloaded. It may work if you try in a different wifi.

**Reply by clairelovesgravy** [reply]
Thank you! after switching a network, it does work! millions of thanks!

**Reply by ReblochonMasque** [reply]
That fixed it for me too, many thanks!

**Comment by NemanSyed** [comment]
Another thread https://github.com/openai/whisper/discussions/1027 indicated it might be due to a bad swap file. The solution was to restart. No permanent solutions were proposed on that thread.

**Comment by fadyazizz** [comment]
on mac I solved this issue by going to /Users/<insert_user_name>/.cache/torch and deleted whisperx-vad-segmentation.bin file
notice that .cache folder is a hidden folder and can be shown by pressing cmd+shift+. 
after that I ran the python program I am working on and the issue is resolved

**Comment by xiongpahao** [comment]
It is completely a network-related issue. You can solve this by manully downloading the model, follow these steps:
1. Find the download link to the model you want, it is listed in /python3.11/site-packages/whisper/__init__.py(if python was installed through homebrew, for example, on macOS, the path is /opt/homebrew/lib/python3.11/site-packages/whisper/__init__.py), the code is:
```python
_MODELS = {
    "tiny.en": "https://openaipublic.azureedge.net/main/whisper/models/d3dd57d32accea0b295c96e26691aa14d8822fac7d9d27d5dc00b4ca2826dd03/tiny.en.pt",
    "tiny": "https://openaipublic.azureedge.net/main/whisper/models/65147644a518d12f04e32d6f3b26facc3f8dd46e5390956a9424a650c0ce22b9/tiny.pt",
    "base.en": "https://openaipublic.azureedge.net/main/whisper/models/25a8566e1d0c1e2231d1c762132cd20e0f96a85d16145c3a00adf5d1ac670ead/base.en.pt",
    "base": "https://openaipublic.azureedge.net/main/whisper/models/ed3a0b6b1c0edf879ad9b11b1af5a0e6ab5db9205f891f668f8b0e6c6326e34e/base.pt",
    "small.en": "https://openaipublic.azureedge.net/main/whisper/models/f953ad0fd29cacd07d5a9eda5624af0f6bcf2258be67c92b79389873d91e0872/small.en.pt",
    "small": "https://openaipublic.azureedge.net/main/whisper/models/9ecf779972d90ba49c06d968637d720dd632c55bbf19d441fb42bf17a411e794/small.pt",
    "medium.en": "https://openaipublic.azureedge.net/main/whisper/models/d7440d1dc186f76616474e0ff0b3b6b879abc9d1a4926b7adfa41db2d497ab4f/medium.en.pt",
    "medium": "https://openaipublic.azureedge.net/main/whisper/models/345ae4da62f9b3d59415adc60127b97c714f32e89e936602e85993674d08dcb1/medium.pt",
    "large-v1": "https://openaipublic.azureedge.net/main/whisper/models/e4b87e7e0bf463eb8e6956e646f1e277e901512310def2c24bf0e11bd3c28e9a/large-v1.pt",
    "large-v2": "https://openaipublic.azureedge.net/main/whisper/models/81f7c96c852ee8fc832187b0132e569d6c3065a3252ed18e56effd0b6a73e524/large-v2.pt",
    "large-v3": "https://openaipublic.azureedge.net/main/whisper/models/e5b1a55b89c1367dacf97e3e19bfd829a01529dbfdeefa8caeb59b3f1b81dadb/large-v3.pt",
    "large": "https://openaipublic.azureedge.net/main/whisper/models/e5b1a55b89c1367dacf97e3e19bfd829a01529dbfdeefa8caeb59b3f1b81dadb/large-v3.pt",
}
```
2. Copy the according download link to a downloader (like Neat Download Manager on macOS), then start to download the model, like large-v3.pt
3. When the download finishes, move the file (large-v3.pt here) to /Users/USER/.cache/whisper/.
Now everything should work fine.

**Reply by ForestWalkUK** [reply]
Thank you, I just downloaded a model using one of the URLs you supplied above and it works on Windows 10. Apparently the model files are cross-platform compatible.

**Reply by xiongpahao** [reply]
Glad to hear that, they are indeed cross-platform compatible.

**Reply by dkoloditch** [reply]
For those using `poetry` on macOS, path may look something like the following `/Users/<yourusername>/Library/Caches/pypoetry/virtualenvs/whisper-7RZr_1iN-py3.11/lib/python3.11/site-packages/whisper/__init__.py`.
